### PR TITLE
Add extra fields to JobStatus Result object

### LINKED ISSUE #501: Job status model is not match with zendesk API response
### **we found JobStatus model from zendeskApi.JobStatuses.GetJobStatusAsync method is not matched response from zendesk job API.** 

image
image

**Here is response from zendesk API by postman:**
image

### PR DESCRIPTION
**Pull request by APErebus**
As per #501, there are number of fields on a JobStatus Result object that aren't currently being deserialized. I have added:

- `index`
- `error`
- `details`

I wasn't 100% sure how to write a test to handle this one. 

fixes #501

### CODE REVIEW

File: src/ZendeskApi_v2/Models/Shared/Result.cs [hunk 1]
```diff
@@ -22,5 +21,14 @@ public class Result
 
         [JsonProperty("status")]
         public string Status { get; set; }
+
+        [JsonProperty("index")]
+        public int Index { get; set; }
```
Comment (mozts2005) [inline src/ZendeskApi_v2/Models/Shared/Result.cs:26]: please make this a long.  see https://developer.zendesk.com/api-reference/ticketing/introduction/#data-types

```suggestion
        public long Index { get; set; }
```

**Review by mozts2005 — CHANGES_REQUESTED** [review]
Just needs a small fix

File: src/ZendeskApi_v2/Models/Shared/Result.cs (hunk 1 above)
Comment (APErebus) [reply src/ZendeskApi_v2/Models/Shared/Result.cs]: Ah derp. Not sure how I missed that. Apologies.

**Comment by mozts2005** [comment]
I will merge this just need to update the account used for testing as Zendesk has asked that I make some changes.

**Review by mozts2005 — APPROVED** [review]

**Pull request merged**